src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/controller/PremisesController.kt

```diff
@@ -150,4 +152,31 @@ class PremisesController(
 
     return ResponseEntity.ok(lostBedsTransformer.transformJpaToApi(lostBed))
   }
+
+  override fun premisesPremisesIdCapacityGet(premisesId: UUID): ResponseEntity<List<DateCapacity>> {
+    val premises = premisesService.getPremises(premisesId)
+      ?: throw NotFoundProblem(premisesId, "Premises")
+
+    val lastBookingDate = premisesService.getLastBookingDate(premises)
+    val lastLostBedsDate = premisesService.getLastLostBedsDate(premises)
+
+    val capacityForPeriod = premisesService.getAvailabilityForRange(
+      premises,
+      LocalDate.now(),
+      maxOf(
+        LocalDate.now(),
+        lastBookingDate ?: LocalDate.now(),
+        lastLostBedsDate ?: LocalDate.now()
+      )
```

Comment:
Love it!

Reply:
elvis operator up there with the best kotlin features :D